go/enclave/components/rollup_consumer.go

```diff
@@ -117,98 +163,73 @@ func (rc *rollupConsumerImpl) ExportAndVerifyCrossChainData(ctx context.Context,
 	return serializedTree, nil
 }
 
-// GetRollupsFromL1Data - extracts the rollups from the processed L1 data and checks sequencer signature on them
-func (rc *rollupConsumerImpl) GetRollupsFromL1Data(processed *common.ProcessedL1Data) ([]*common.ExtRollup, error) {
-	defer core.LogMethodDuration(rc.logger, measure.NewStopwatch(), "Rollup consumer get rollups from L1 data", log.BlockHashKey, processed.BlockHeader.Hash())
-
-	block := processed.BlockHeader
-	rollups, err := rc.extractAndVerifyRollups(processed)
-	if err != nil {
-		rc.logger.Error("Failed to extract rollups from block", log.BlockHashKey, block.Hash(), log.ErrKey, err)
-		return nil, err // if multiple rollups are found with the same tx hash we will return here
-	}
-	if len(rollups) == 0 {
-		rc.logger.Trace("No rollups found in block", log.BlockHashKey, block.Hash())
-		return nil, nil
+// verifySequencerSignature - verifies that a rollup transaction was properly signed by the sequencer.
+//
+// 1. Extracts blobs and signatures from the transaction
+// 2. Computes blob hashes using KZG commitments
+// 3. Reconstructs the rollup from blob data
+// 4. Verifies the sequencer signature using a composite hash of the rollup header and blob hash
+//
+// Note: All errors are considered non-critical as they occur prior to signature verification
+// and could be due to malformed or invalid input data. We don't want to prevent blocks from being processed if this is
+// the case.
+func (rc *rollupConsumerImpl) verifySequencerSignature(rollupTx *common.L1TxData) (*common.ExtRollup, []gethcommon.Hash, error) {
```

Comment:
feels like this has too many responsibilites. What about splitting it in a method that produces the ExtRollup, and a method that checks the signature